### PR TITLE
chore: update deps

### PR DESCRIPTION
**Pull request by nacardin**
* Update all depencencies to latest versions
* Timer sleeper only fires once
* Remove `http-client`
* Rename `native2_tls` feature to `native_tls`
* Note: `rustls` now uses `aws-lc` instead of `ring`

Includes https://github.com/infinyon/future-aio/pull/241 and https://github.com/infinyon/future-aio/pull/258

Resolves https://github.com/infinyon/future-aio/issues/239

### CODE REVIEW

**Comment by morenol** [comment]
@nacardin maybe this PR could be helpful https://github.com/infinyon/future-aio/pull/258

**Comment by nacardin** [comment]
> @nacardin maybe this PR could be helpful #258

That seems to change behavior to only fire once. I made a change to preserve existing behavior of firing the timer for every await of the future.

**Comment by morenol** [comment]
I reverted it because @sehz mentioned me that it should only fire once:

https://github.com/infinyon/future-aio/pull/258#discussion_r1595669083

**Review by morenol — APPROVED** [review]

File: fluvio-future-derive/Cargo.toml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 [package]
-name = "fluvio-test-derive"
-version = "0.1.1"
-edition = "2018"
+name = "fluvio-future-derive"
+version = "0.1.0"
+edition = "2021"
```
Comment (sehz) [inline fluvio-future-derive/Cargo.toml:4]: is there reason this go back to 0.1.0? shouldn't this be 0.1.11 or 0.2

File: src/rust_tls.rs [hunk 2]
```diff
@@ -42,38 +42,49 @@ mod cert {
...
-    use async_rustls::rustls::RootCertStore;
+    use futures_rustls::rustls::pki_types::CertificateDer;
+    use futures_rustls::rustls::pki_types::PrivateKeyDer;
+    use futures_rustls::rustls::RootCertStore;
     use rustls_pemfile::certs;
-    use rustls_pemfile::rsa_private_keys;
+    use rustls_pemfile::pkcs8_private_keys;
 
-    pub fn load_certs<P: AsRef<Path>>(path: P) -> Result<Vec<Certificate>, IoError> {
+    pub fn load_certs<P: AsRef<Path>>(path: P) -> Result<Vec<CertificateDer<'static>>, IoError> {
```
Comment (sehz) [inline src/rust_tls.rs:51]: since we are bump up major version, let's migrate this to anyhow

File: fluvio-future-derive/Cargo.toml (hunk 1 above)
Comment (nacardin) [reply fluvio-future-derive/Cargo.toml:4]: I changed the crate name. I added a main proc macro to it, so its not only test macros anymore.

File: src/rust_tls.rs [hunk 3]
```diff
@@ -42,38 +42,49 @@ mod cert {
...
         load_certs_from_reader(&mut BufReader::new(File::open(path)?))
     }
 
-    pub fn load_certs_from_reader(rd: &mut dyn BufRead) -> Result<Vec<Certificate>, IoError> {
+    pub fn load_certs_from_reader(
+        rd: &mut dyn BufRead,
+    ) -> Result<Vec<CertificateDer<'static>>, IoError> {
```
Comment (sehz) [inline src/rust_tls.rs:57]: move to anyhow

File: src/rust_tls.rs [hunk 4]
```diff
@@ -42,38 +42,49 @@ mod cert {
...
-            .map(|v| v.into_iter().map(Certificate).collect())
-            .map_err(|_| IoError::new(ErrorKind::InvalidInput, "invalid cert"))
+            .map(|r| r.map_err(|_| IoError::new(ErrorKind::InvalidInput, "invalid cert")))
+            .collect()
     }
 
     /// Load the passed keys file
-    pub fn load_keys<P: AsRef<Path>>(path: P) -> Result<Vec<PrivateKey>, IoError> {
+    pub fn load_keys<P: AsRef<Path>>(path: P) -> Result<Vec<PrivateKeyDer<'static>>, IoError> {
         load_keys_from_reader(&mut BufReader::new(File::open(path)?))
```
Comment (sehz) [inline src/rust_tls.rs:65]: anyhow

File: src/rust_tls.rs [hunk 5]
```diff
@@ -42,38 +42,49 @@ mod cert {
...
     }
 
-    pub fn load_keys_from_reader(rd: &mut dyn BufRead) -> Result<Vec<PrivateKey>, IoError> {
-        rsa_private_keys(rd)
-            .map(|v| v.into_iter().map(PrivateKey).collect())
-            .map_err(|_| IoError::new(ErrorKind::InvalidInput, "invalid key"))
+    pub fn load_keys_from_reader(
+        rd: &mut dyn BufRead,
+    ) -> Result<Vec<PrivateKeyDer<'static>>, IoError> {
```
Comment (sehz) [inline src/rust_tls.rs:70]: anyhow

File: src/rust_tls.rs [hunk 6]
```diff
@@ -253,22 +271,23 @@ mod builder {
         pub fn load_ca_cert<P: AsRef<Path>>(
             self,
             path: P,
-        ) -> Result<ConnectorBuilderStage<WantsTransparencyPolicyOrClientCert>, IoError> {
+        ) -> Result<ConnectorBuilderStage<WantsClientCert>, IoError> {
```
Comment (sehz) [inline src/rust_tls.rs:274]: move to anyhow

File: src/rust_tls.rs [hunk 7]
```diff
@@ -277,8 +296,8 @@ mod builder {
 
         fn with_root_certificates(
             self,
-            certs: &[Certificate],
-        ) -> Result<ConnectorBuilderStage<WantsTransparencyPolicyOrClientCert>, IoError> {
+            certs: Vec<CertificateDer>,
+        ) -> Result<ConnectorBuilderStage<WantsClientCert>, IoError> {
```
Comment (sehz) [inline src/rust_tls.rs:300]: anyhow

File: src/rust_tls.rs [hunk 8]
```diff
@@ -395,21 +418,48 @@ mod builder {
...
+            _end_entity: &CertificateDer,
+            _intermediates: &[CertificateDer],
             _server_name: &ServerName,
-            _scts: &mut dyn Iterator<Item = &[u8]>,
             _ocsp_response: &[u8],
-            _now: SystemTime,
+            _now: UnixTime,
         ) -> Result<ServerCertVerified, TlsError> {
-            log::debug!("ignoring server cert");
+            debug!("ignoring server cert");
```
Comment (sehz) [inline src/rust_tls.rs:433]: change to info

File: src/rust_tls.rs [hunk 9]
```diff
@@ -395,21 +418,48 @@ mod builder {
...
             Ok(ServerCertVerified::assertion())
         }
+
+        fn verify_tls12_signature(
+            &self,
+            _message: &[u8],
+            _cert: &CertificateDer<'_>,
+            _dss: &futures_rustls::rustls::DigitallySignedStruct,
+        ) -> Result<HandshakeSignatureValid, TlsError> {
+            debug!("ignoring server cert");
```
Comment (sehz) [inline src/rust_tls.rs:443]: info log

File: fluvio-future-derive/Cargo.toml (hunk 1 above)
Comment (sehz) [reply fluvio-future-derive/Cargo.toml:4]: got it

**Review by sehz — CHANGES_REQUESTED** [review]
Looks good.  few minor nits on changing to anyhow and log level

File: src/rust_tls.rs (hunk 2 above)
Comment (nacardin) [reply src/rust_tls.rs]: done, looks a lot cleaner with anyhow

**Review by sehz — APPROVED** [review]
Nice.  LGTM

**Comment by nacardin** [comment]
@sehz We should also convert `native_tls` and `openssl_tls` errors to anyhow?

**Pull request merged**